crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/mod.rs

```diff
@@ -171,6 +172,10 @@ fn execute_syscall<Request: ExecuteCall + SyscallRequest>(
     Ok(())
 }
 
+pub struct RuntimeState<'a> {
+    pub cheatnet_state: &'a mut CheatnetState,
+}
```

Comment:
Is it good that it has a reference to the state instead of owning it? If you wanna add `ObserverState` later on we then will have this weird struct owning only part of the state and it may be harder to satisfy borrow checker in the future

Reply:
we cannot own it because we have to share it between multiple runtimes.